File: src/main/java/icu/takeneko/omms/client/data/system/NetworkInfo.java
package icu.takeneko.omms.client.data.system;

import com.google.gson.GsonBuilder;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@AllArgsConstructor
public class NetworkInfo {
    private final List<NetworkInterface> networkInterfaceList = new ArrayList<>();
    private final String hostName;
    private final String domainName;
    private final String[] dnsServers;
    private final String ipv4DefaultGateway;
    private final String ipv6DefaultGateway;

    public static String toJsonString(NetworkInfo networkInfo) {
        return new GsonBuilder().serializeNulls().create().toJson(networkInfo);
    }

    @Getter
    @AllArgsConstructor
    public static class NetworkInterface {
        String name;
        String displayName;
        String macAddress;
        long mtu;
        long speed;
        String[] ipv4Address;
        String[] ipv6Address;
    }
}

File: src/main/java/icu/takeneko/omms/client/data/system/NetworkInfo.kt
package icu.takeneko.omms.client.data.system

data class NetworkInfo(
    var hostName: String,
    var domainName: String,
    var dnsServers: List<String>,
    var ipv4DefaultGateway: String,
    var ipv6DefaultGateway: String,
    var networkInterfaceList: List<NetworkInterface> = listOf(),
)

data class NetworkInterface(
    var name: String,
    var displayName: String,
    var macAddress: String,
    var mtu: Long,
    var speed: Long,
    var ipv4Address: List<String>,
    var ipv6Address: List<String>,
)
